### ISSUE TITLE
Echo hangs and loops infinitely

### ISSUE THREAD

**Issue by tpizza** [robrich/gulp-exec#30]
If I echo out to a file, the command runs, but it loops infinitely and recursively writes to the target file.

Here's my use-case as to why I'm even performing an echo:

I really enjoy the safeguard in password protecting my SSH key.  Not so much for the extra 'security', but it gives me a brief moment to decide if there's something I forgot and back out of a push.  I'm in the process of automating an upstream setup using Gulp where I need to use a password-less key to handle the subtree setup.  The point of the echo is to write to the SSH config to change the key to the p/w-less one for setting, pulling and merging the subtree, then restoring it once the process completes.

This is all you need to reproduce:

`GULP.task('_homez', () => {
    return GULP.src('./**/**')
        .pipe( PLUGINS.exec('echo test > test.txt') )
        .pipe( PLUGINS.exec.reporter());
} );`

**Comment by robrich** [comment]
`./**/**` grabs everything including `node_modules`. It ran fine for me,
but took a long time. Switch to `echo test >> test.txt` to see how many
times it ran. Add a `!./node_modules/**/**` glob or otherwise limit the
input file list so it'll do less work.

**Comment by tpizza** [comment]
Interesting.  When I switched the command to append the echoes, I got to about 8 tests until I killed the task.  For a second I thought maybe it was because the task was trying to also run through the IntelliJ project folder, but it would still hang even with `!./.idea/**/**`.  I got it working, however instead of the wildcard glob, I passed src the project root `./` and the task would successfully finish.
